kernel/src/fs/procfs/pid/task/status.rs

```diff
@@ -105,8 +105,7 @@ impl FileOps for StatusFileOps {
             posix_thread
                 .thread_name()
                 .lock()
-                .as_ref()
-                .and_then(|name| name.as_string())
+                .as_string()
```

Comment:
Can we remove the `as_string` API and try to write the `CStr` instead? `as_string` does not seem to make much sense now especially `ThreadName` cannot be empty.